app/src/org/commcare/android/javarosa/IntentCallout.java

```diff
@@ -303,10 +309,53 @@ private void storePointerToFileResponse(TreeReference ref, String responseValue,
         if (!src.exists()) {
             //TODO: How hard should we be failing here?
             Log.w(TAG, "CommCare received a link to a file at " + src.toString() + " to be included in the form, but it was not present on the phone!");
+
+            Uri uri = Uri.parse(responseValue);
+
+            try {
+                ParcelFileDescriptor inFile = androidContext.getContentResolver().openFileDescriptor(uri, "r");
```

Comment:
`getContentResolver()` works fine with application context. We can avoid the above changes to method signatures by using application context here.

Reply:
Ah, that's a good point I hadn't really considered. In general I've always considered it a bit of an antipattern to use the ApplicationContext, but given the amount of overhead associated with bubbling down the activity specific context in this particular case, and the fact that we're just using it to get a generic content resolver, I could definitely see why it might make more sense here.